pkg/cmd/adm/restart.go

```diff
@@ -3,155 +3,198 @@ package adm
+		In:     os.Stdin,
+		Out:    os.Stdout,
+		ErrOut: os.Stderr,
+	}
+	kubeConfigFlags.ClusterName = nil  // `cluster` flag is redefined for our own purpose
+	kubeConfigFlags.AuthInfoName = nil // unused here, so we can hide it
+	kubeConfigFlags.Context = nil      // unused here, so we can hide it
+
 	cfg, err := configuration.LoadClusterConfig(ctx, clusterName)
 	if err != nil {
 		return err
 	}
-	cl, err := ctx.NewClient(cfg.Token, cfg.ServerAPI)
+	kubeConfigFlags.Namespace = &cfg.OperatorNamespace
+	kubeConfigFlags.APIServer = &cfg.ServerAPI
+	kubeConfigFlags.BearerToken = &cfg.Token
+	kubeconfig, err := client.EnsureKsctlConfigFile()
 	if err != nil {
 		return err
 	}
+	kubeConfigFlags.KubeConfig = &kubeconfig
 
-	if len(deployments) == 0 {
-		err := printExistingDeployments(ctx.Terminal, cl, cfg.OperatorNamespace)
-		if err != nil {
-			ctx.Terminal.Printlnf("\nERROR: Failed to list existing deployments\n :%s", err.Error())
-		}
-		return fmt.Errorf("at least one deployment name is required, include one or more of the above deployments to restart")
+	cl, err := ctx.NewClient(cfg.Token, cfg.ServerAPI)
+	if err != nil {
+		return err
 	}
-	deploymentName := deployments[0]
 
 	if !ctx.AskForConfirmation(
-		ioutils.WithMessagef("restart the deployment '%s' in namespace '%s'", deploymentName, cfg.OperatorNamespace)) {
+		ioutils.WithMessagef("restart all the deployments in the cluster  '%s' and namespace '%s' \n", clusterName, cfg.OperatorNamespace)) {
 		return nil
 	}
-	return restartDeployment(ctx, cl, cfg.OperatorNamespace, deploymentName)
+
+	return restartDeployment(ctx, cl, cfg.OperatorNamespace, factory, ioStreams)
 }
 
-func restartDeployment(ctx *clicontext.CommandContext, cl runtimeclient.Client, ns string, deploymentName string) error {
-	namespacedName := types.NamespacedName{
-		Namespace: ns,
-		Name:      deploymentName,
-	}
+func restartDeployment(ctx *clicontext.CommandContext, cl runtimeclient.Client, ns string, f cmdutil.Factory, ioStreams genericclioptions.IOStreams) error {
+	fmt.Printf("Fetching the current OLM and non-OLM deployments of the operator in %s \n", ns)
 
-	originalReplicas, err := scaleToZero(cl, namespacedName)
+	olmDeploymentList, nonOlmDeploymentlist, err := getExistingDeployments(cl, ns)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			ctx.Printlnf("\nERROR: The given deployment '%s' wasn't found.", deploymentName)
-			return printExistingDeployments(ctx, cl, ns)
-		}
-		return err
-	}
-	ctx.Println("The deployment was scaled to 0")
-	if err := scaleBack(ctx, cl, namespacedName, originalReplicas); err != nil {
-		ctx.Printlnf("Scaling the deployment '%s' in namespace '%s' back to '%d' replicas wasn't successful", originalReplicas)
-		ctx.Println("Please, try to contact administrators to scale the deployment back manually")
 		return err
 	}
 
-	ctx.Printlnf("The deployment was scaled back to '%d'", originalReplicas)
+	if len(olmDeploymentList.Items) == 0 {
```

Comment:
For a better user experience should we let users know that there was 0 deployment hence we did not go ahead with restart? and not go through the rest of the code ?

Reply:
I think locally it doesn't change much, having an else and printing that there were no deployments or printing the number of deployments found at the beginning and going throw the rest of the code which will just skip the for loops since there will be no items to iterate on. But I'm fine with whatever looks more readable.